packages/guides/src/ReferenceResolvers/AnchorReferenceResolver.php

```diff
@@ -24,7 +25,12 @@ public function __construct(
     public function resolve(LinkInlineNode $node, RenderContext $renderContext): bool
     {
         $reducedAnchor = $this->anchorReducer->reduceAnchor($node->getTargetReference());
-        $target = $renderContext->getProjectNode()->getInternalTarget($reducedAnchor);
+        if ($node instanceof ReferenceNode) {
+            $target = $renderContext->getProjectNode()->getInternalTarget($reducedAnchor, $node->getLinkType());
+        } else {
+            $target = $renderContext->getProjectNode()->getInternalTarget($reducedAnchor);
+        }
+
```

Comment:
It feels a bit odd to have a else in here, as we already have a way to find the correct Reference resolvers. Should this not be a different strategy? maybe with some shared base class, but right now this looks like a strategy inside a strategy. 

If I look at the example of the in phpDocumentor introduced link targets, I use the domain to check if the domain is `php` if it's not a known `php` link I ignore the link and pass it on to the next resolver. I think this should also be the case for other `LinkTypes`.

Reply:
I added a todo, we can solve this later